daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dsom/ChoiceGroup.scala

```diff
@@ -257,3 +275,13 @@ final class Choice(xmlArg: Node, parent: SchemaComponent, position: Int)
       optIgnoreCase)
   }
 }
+
+final class Choice(xmlArg: Node, parent: SchemaComponent, position: Int)
+  extends ChoiceBase(xmlArg, parent, position)
+  with ChoiceLikeMixin
```

Comment:
If ChoiceBase should mixin ChoiceLikeMixin, this isn't necessary. But maybe the above is wrong?

Reply:
ChoiceBase should not mixin ChoiceLikeMixin, as ChoiceLikeMixin is for things that *define* choice groups, and ChoiceBase is base for terms that act like choice groups at runtime. 
ChoiceBase is base of ChoiceRefs as well as Choice. ChoiceRefs don't define a choice group.  

This is why I think I should rename ChoiceLikeMixin to ChoiceDefMixin.